### PR TITLE
Fix 3167 (package downgrade when referencing IdentityModel.Tokens from dev)

### LINKED ISSUE #3167: IdWeb fails with current version of IdentityModel dev branch
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

3,5,1

### Web app

Not Applicable

### Web API

Not Applicable

### Token cache serialization

Not Applicable

### Description

IdWeb fails with current version of IdentityModel dev branch, due to a package downgrade (Of Microsoft.Extensions.Logging.Abstractions)

### Reproduction steps

1. Run a unified build with all master/main/dev versions of the libraries

### Error message

D:\a_work\1\s\microsoft-identity-web\src\Microsoft.Identity.Web.Certificateless\Microsoft.Identity.Web.Certificateless.csproj : error NU1605: Warning As Error: Detected package downgrade: Microsoft.Extensions.Logging.Abstractions from 3.1.0 to 2.1.0. Reference the package directly from the project to select a different version. [D:\a_work\1\s\microsoft-identity-web\tests\E2E Tests\NET 7 tests\IntegrationTests\IntegrationTests.csproj]
D:\a_work\1\s\microsoft-identity-web\src\Microsoft.Identity.Web.Certificateless\Microsoft.Identity.Web.Certificateless.csproj : error NU1605: Microsoft.Identity.Web.Certificateless -> Microsoft.IdentityModel.JsonWebTokens 8.3.1 -> Microsoft.IdentityModel.Tokens 8.3.1 -> Microsoft.Extensions.Logging.Abstractions (>= 3.1.0) [D:\a_work\1\s\microsoft-identity-web\tests\E2E Tests\NET 7 tests\IntegrationTests\IntegrationTests.csproj]
D:\a_work\1\s\microsoft-identity-web\src\Microsoft.Identity.Web.Certificateless\Microsoft.Identity.Web.Certificateless.csproj : error NU1605: Microsoft.Identity.Web.Certificateless -> Microsoft.Extensions.Logging.Abstractions (>= 2.1.0) [D:\a_work\1\s\microsoft-identity-web\tests\E2E Tests\NET 7 tests\IntegrationTests\IntegrationTests.csproj]

### Id Web logs

N/A

### Relevant code snippets

```csharp
N/a
```

### Regression

No regression

### Expected behavior

IdWeb builds.

Analysis. This is because IdentityModel now leverages Microsoft.Extensions.Loging.Abstraction

### PR DESCRIPTION
**Pull request by jmprieur**
# Fix 3167 (package downgrade when referencing IdentityModel.Tokens from dev)

Removed now un-needed reference

Fixes #3167

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: Directory.Build.props [hunk 1]
```diff
@@ -161,19 +161,22 @@
...
     <SystemSecurityCryptographyXmlVersion>6.0.1</SystemSecurityCryptographyXmlVersion>
 
     <!-- CVE-2022-34716 due to DataProtection 5.0.8 -->
-    <MicrosoftExtensionsLoggingVersion>5.0.0</MicrosoftExtensionsLoggingVersion>
     <SystemTextEncodingsWebVersion>6.0.0</SystemTextEncodingsWebVersion>
 
+    <!-- 6.0.0 as 5.x are deprecated -->
+    <MicrosoftExtensionsLoggingVersion>6.0.0</MicrosoftExtensionsLoggingVersion>
+
+
```
Comment (brentschmaltz) [inline Directory.Build.props:179]: perhaps remove this line?

**Review by brentschmaltz — APPROVED** [review]
nit; extra line.

**Review by pmaytak — APPROVED** [review]

**Comment by pmaytak** [comment]
@jmprieur Error was because Identity Web was downgrading the package (because of 5x reference) and IdentityModel was referencing the newer one?

File: Directory.Build.props
Comment (pmaytak) [inline Directory.Build.props:1]: Do the same changes need to be made for net472 and netstandard2?

https://github.com/AzureAD/microsoft-identity-web/blob/3cdec89d56f313ca8953e3df2ab0c0ac1bb0a325/Directory.Build.props#L193

File: Directory.Build.props (hunk 1 above)
Comment (jmprieur) [reply Directory.Build.props]: done

**Comment by jmprieur** [comment]
> @jmprieur The error happened because Identity Web was the one downgrading the package (because of 5x reference) and IdentityModel was referencing the newer one?

It's happening because CredentialLess exposes only netstandard2.0 -> references Microsoft.Extensions.Logging.Abstractions 2.1 through Mirosoft.IdentityModel..Tokens, but then Credentials exposes all the target frameworks, and therefore we needed to re-reference the right Microsoft.Extensions.Logging.Abstractions version per framework, do avoid the package downgrade in other assemblies.

File: Directory.Build.props
Comment (jmprieur) [reply Directory.Build.props:1]: I don't understand the question?

Comment (pmaytak) [reply Directory.Build.props:1]: Do we need to upgrade these Extensions references to 6.x also?

Comment (jmprieur) [reply Directory.Build.props:1]: no, 2.1.0 is fine

image


the 5.x are not:

image

**Pull request merged**